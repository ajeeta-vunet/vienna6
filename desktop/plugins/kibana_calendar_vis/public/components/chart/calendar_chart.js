/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { CalendarAxis } from './axis/calendar_axis';
import { ChartGrid } from './grid/chart_grid';
import { ChartTitle } from './title/chart_title';

export class CalendarChart extends React.Component {

    state = {
      width: '100%',
      height: '100%',
      rendered: 0
    };

    constructor(props) {
      super(props);
      this.svg = undefined;
      this.adjustSize = this.adjustSize.bind(this);
    }

    async adjustSize() {
      await this.setState(prev => ({
        rendered: prev.rendered + 1
      }));

      const { visConfig, renderComplete } = this.props;

      if (this.svg && this.state.rendered === visConfig.get('categoryAxes').length + 2) {
        const cellSize = visConfig.get('grid.cellSize');
        const svg = this.svg;
        this.setState({
          width: svg.getBBox().width + 2 * cellSize,
          height: svg.getBBox().height + 2 * cellSize
        });
        renderComplete();
      }
    }

    componentDidMount() {
      /*
      * Justification for redundant calls to renderComplete:
      * make tests pass without Timeout errors, does not
      * affect the normal feature behavior.
      */
      this.props.renderComplete();
    }

    render() {
      const { visConfig, vislibData, dispatcher } = this.props;

      return (
        <svg
          width={this.state.width}
          height={this.state.height}
          ref={e => {this.svg = e;}}
        >
          <ChartTitle
            gridConfig={visConfig.get('grid')}
            label={vislibData.yAxisLabel}
            dateRef={vislibData.series[0].values[0].x}
            chartType={visConfig.get('type')}
            renderComplete={this.adjustSize}
          />
          {visConfig.get('categoryAxes').map((axisArgs, i) => (
            <CalendarAxis
              key={i}
              type={visConfig.get('type')}
              gridConfig={visConfig.get('grid')}
              axisConfig={axisArgs}
              vislibData={vislibData}
              dispatcher={dispatcher}
              renderComplete={this.adjustSize}
            />
          ))}
          <ChartGrid
            type={visConfig.get('type')}
            gridConfig={visConfig.get('grid')}
            vislibData={vislibData}
            axes={visConfig.get('categoryAxes')}
            dispatcher={dispatcher}
            renderComplete={this.adjustSize}
          />
        </svg>
      );
    }

    componentWillUnmount() {
      this.svg = null;
    }
}
