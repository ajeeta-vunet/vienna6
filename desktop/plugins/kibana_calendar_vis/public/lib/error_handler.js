/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import MarkdownIt from 'markdown-it';
import d3 from 'd3';

const markdownIt = new MarkdownIt({
  html: true,
  linkify: true
});

export const errorContainer = 'visualize-error';

export class CalendarErrorHandler {
  constructor() {
    this.el = undefined;
    this.id = undefined;
  }

  bindEl(el, id) {
    this.el = el;
    this.id = id;
    return this;
  }

  removeError() {
    d3.select(this.el)
      .attr('id', this.id)
      .selectAll(`.${errorContainer}`)
      .remove();
  }

  error(msg) {
    this.removeError();
    const div = d3.select(this.el)
      .append('div')
      .attr('id', this.id)
      // class name needs `chart` in it for the polling checkSize function
      // to continuously call render on resize
      .attr('class', `${errorContainer} chart error`);

    if (msg === 'No results found') {
      div.append('div')
        .attr('class', `text-center ${errorContainer} visualize-chart`)
        .append('div').attr('class', 'item top')
        .append('div').attr('class', 'item')
        .append('h3').html('No data to show');

      div.append('div').attr('class', 'item bottom');
    } else {
      div.append('h4').html(markdownIt.renderInline(msg));
    }
    return div;
  }
}
