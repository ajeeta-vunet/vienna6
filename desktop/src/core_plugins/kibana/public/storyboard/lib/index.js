export { saveStoryboard } from './save_storyboard';
//export { getAppStateDefaults } from './get_app_state_defaults';
