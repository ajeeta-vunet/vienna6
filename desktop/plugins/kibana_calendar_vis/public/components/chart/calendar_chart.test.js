/* eslint-disable */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import { CalendarChart } from './calendar_chart';
import { CalendarVisConfig, Dispatcher } from '../../lib';
import { defaultParams, monthViewParams, dayViewParams } from '../../default_settings';
import aggResponse from '../../__tests__/agg_response.json';
import { truncateUnusable, replicateDate } from '../../../test/jest/utils';
import { calculateBounds } from 'ui/timefilter/get_time';

describe('CalendarChart', () => {

  let visConfig;
  let visData;
  let dispatcher;
  const fakeVis = {
    params: defaultParams,
    API: {
      timeFilter: {
        getBounds: function () {
          return calculateBounds({
            from: moment(1531026000000),
            to: moment(1531026000000 + 35 * 86400000)
          });
        }
      }
    }
  };

  beforeEach(() => {
    dispatcher = new Dispatcher().addContainer(document.createElement('div'));
    visData = truncateUnusable(aggResponse.rows[0]);
  });

  afterEach(() => {
    visConfig = null;
    visData = null;
  });

  it('should render a year view chart with two category axes, one grid and a title', () => {
    visConfig = new CalendarVisConfig(fakeVis, defaultParams);
    // replicate data
    visData = replicateDate(visData);

    const renderComplete = jest.fn();
    const chartWrapper = shallow(<CalendarChart
      id={`chart_${visData.label.slice(0, 4)}`}
      visConfig={visConfig}
      vislibData={visData}
      dispatcher={dispatcher}
      renderComplete={renderComplete}
    />);
    expect(chartWrapper).toMatchSnapshot();
  });

  it('should render a month view chart with one category axis, one grid and a title', () => {
    visConfig = new CalendarVisConfig(fakeVis, monthViewParams);
    // replicate data
    visData = replicateDate(visData, 3);

    const renderComplete = jest.fn();
    const chartWrapper = shallow(<CalendarChart
      id={`chart_${visData.label.slice(0, 4)}`}
      visConfig={visConfig}
      vislibData={visData}
      dispatcher={dispatcher}
      renderComplete={renderComplete}
    />);
    expect(chartWrapper).toMatchSnapshot();
  });

  it('should render a day view chart with two category axes, one grid and a title', () => {
    visConfig = new CalendarVisConfig(fakeVis, dayViewParams);
    // replicate data
    visData = replicateDate(visData, 0);

    const renderComplete = jest.fn();
    const chartWrapper = shallow(<CalendarChart
      id={`chart_${visData.label.slice(0, 4)}`}
      visConfig={visConfig}
      vislibData={visData}
      dispatcher={dispatcher}
      renderComplete={renderComplete}
    />);
    expect(chartWrapper).toMatchSnapshot();
  });
});
