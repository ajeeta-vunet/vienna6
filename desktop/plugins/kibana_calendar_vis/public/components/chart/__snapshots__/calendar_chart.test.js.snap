// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarChart should render a day view chart with two category axes, one grid and a title 1`] = `
<svg
  height="100%"
  width="100%"
>
  <ChartTitle
    chartType="heatmap_day"
    dateRef={1531026000000}
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    label="Count"
    renderComplete={[Function]}
  />
  <CalendarAxis
    axisConfig={
      Object {
        "id": "CategoryAxis-1",
        "position": "top",
        "scale": Object {
          "type": "HOURS",
        },
        "type": "category",
      }
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    key="0"
    renderComplete={[Function]}
    type="heatmap_day"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
  <CalendarAxis
    axisConfig={
      Object {
        "id": "CategoryAxis-2",
        "position": "left",
        "scale": Object {
          "type": "MERIDIEM",
        },
        "type": "category",
      }
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    key="1"
    renderComplete={[Function]}
    type="heatmap_day"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
  <ChartGrid
    axes={
      Array [
        Object {
          "id": "CategoryAxis-1",
          "position": "top",
          "scale": Object {
            "type": "HOURS",
          },
          "type": "category",
        },
        Object {
          "id": "CategoryAxis-2",
          "position": "left",
          "scale": Object {
            "type": "MERIDIEM",
          },
          "type": "category",
        },
      ]
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    renderComplete={[Function]}
    type="heatmap_day"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
</svg>
`;

exports[`CalendarChart should render a month view chart with one category axis, one grid and a title 1`] = `
<svg
  height="100%"
  width="100%"
>
  <ChartTitle
    chartType="heatmap_month"
    dateRef={1531026000000}
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    label="Count"
    renderComplete={[Function]}
  />
  <CalendarAxis
    axisConfig={
      Object {
        "id": "CategoryAxis-1",
        "position": "top",
        "scale": Object {
          "type": "WEEKS",
        },
        "type": "category",
      }
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    key="0"
    renderComplete={[Function]}
    type="heatmap_month"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531112400000,
                "y": 0,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531198800000,
                "y": 1,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531285200000,
                "y": 2,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
  <ChartGrid
    axes={
      Array [
        Object {
          "id": "CategoryAxis-1",
          "position": "top",
          "scale": Object {
            "type": "WEEKS",
          },
          "type": "category",
        },
      ]
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 40,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    renderComplete={[Function]}
    type="heatmap_month"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531112400000,
                "y": 0,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531198800000,
                "y": 1,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531285200000,
                "y": 2,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
</svg>
`;

exports[`CalendarChart should render a year view chart with two category axes, one grid and a title 1`] = `
<svg
  height="100%"
  width="100%"
>
  <ChartTitle
    chartType="heatmap_year"
    dateRef={1531026000000}
    gridConfig={
      Object {
        "cellSize": 15,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    label="Count"
    renderComplete={[Function]}
  />
  <CalendarAxis
    axisConfig={
      Object {
        "id": "CategoryAxis-1",
        "position": "top",
        "scale": Object {
          "type": "MONTHS",
        },
        "type": "category",
      }
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 15,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    key="0"
    renderComplete={[Function]}
    type="heatmap_year"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531112400000,
                "y": 0,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531198800000,
                "y": 1,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531285200000,
                "y": 2,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531371600000,
                "y": 3,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531458000000,
                "y": 4,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531544400000,
                "y": 5,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531630800000,
                "y": 6,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531717200000,
                "y": 7,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531803600000,
                "y": 8,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531890000000,
                "y": 9,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531976400000,
                "y": 10,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532062800000,
                "y": 11,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532149200000,
                "y": 12,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532235600000,
                "y": 13,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532322000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532408400000,
                "y": 15,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532494800000,
                "y": 16,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532581200000,
                "y": 17,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532667600000,
                "y": 18,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532754000000,
                "y": 19,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532840400000,
                "y": 20,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532926800000,
                "y": 21,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533013200000,
                "y": 22,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533099600000,
                "y": 23,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533186000000,
                "y": 24,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533272400000,
                "y": 25,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533358800000,
                "y": 26,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533445200000,
                "y": 27,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533531600000,
                "y": 28,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533618000000,
                "y": 29,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533704400000,
                "y": 30,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533790800000,
                "y": 31,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533877200000,
                "y": 32,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533963600000,
                "y": 33,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1534050000000,
                "y": 34,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
  <CalendarAxis
    axisConfig={
      Object {
        "id": "CategoryAxis-2",
        "position": "left",
        "scale": Object {
          "type": "WEEKS",
        },
        "type": "category",
      }
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 15,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    key="1"
    renderComplete={[Function]}
    type="heatmap_year"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531112400000,
                "y": 0,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531198800000,
                "y": 1,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531285200000,
                "y": 2,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531371600000,
                "y": 3,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531458000000,
                "y": 4,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531544400000,
                "y": 5,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531630800000,
                "y": 6,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531717200000,
                "y": 7,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531803600000,
                "y": 8,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531890000000,
                "y": 9,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531976400000,
                "y": 10,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532062800000,
                "y": 11,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532149200000,
                "y": 12,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532235600000,
                "y": 13,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532322000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532408400000,
                "y": 15,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532494800000,
                "y": 16,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532581200000,
                "y": 17,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532667600000,
                "y": 18,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532754000000,
                "y": 19,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532840400000,
                "y": 20,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532926800000,
                "y": 21,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533013200000,
                "y": 22,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533099600000,
                "y": 23,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533186000000,
                "y": 24,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533272400000,
                "y": 25,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533358800000,
                "y": 26,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533445200000,
                "y": 27,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533531600000,
                "y": 28,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533618000000,
                "y": 29,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533704400000,
                "y": 30,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533790800000,
                "y": 31,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533877200000,
                "y": 32,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533963600000,
                "y": 33,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1534050000000,
                "y": 34,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
  <ChartGrid
    axes={
      Array [
        Object {
          "id": "CategoryAxis-1",
          "position": "top",
          "scale": Object {
            "type": "MONTHS",
          },
          "type": "category",
        },
        Object {
          "id": "CategoryAxis-2",
          "position": "left",
          "scale": Object {
            "type": "WEEKS",
          },
          "type": "category",
        },
      ]
    }
    dispatcher={
      Dispatcher {
        "API": null,
        "config": null,
        "container": <div />,
        "dataTarget": null,
        "src": null,
      }
    }
    gridConfig={
      Object {
        "cellSize": 15,
        "style": "rect",
        "xOffset": 20,
        "yOffset": 20,
      }
    }
    renderComplete={[Function]}
    type="heatmap_year"
    vislibData={
      Object {
        "indexPattern": "c85d40c0-837f-11e8-aa0a-c1e0f6dce300",
        "label": "2018: @timestamp per year",
        "ordered": Object {
          "date": true,
          "interval": "P1D",
          "max": "2018-07-09T14:14:22.521Z",
          "min": "2018-07-08T05:00:00.000Z",
        },
        "series": Array [
          Object {
            "aggId": "1",
            "aggLabel": "Count",
            "count": 0,
            "label": "Count",
            "values": Array [
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531026000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531112400000,
                "y": 0,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531198800000,
                "y": 1,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531285200000,
                "y": 2,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531371600000,
                "y": 3,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531458000000,
                "y": 4,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531544400000,
                "y": 5,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531630800000,
                "y": 6,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531717200000,
                "y": 7,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531803600000,
                "y": 8,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531890000000,
                "y": 9,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1531976400000,
                "y": 10,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532062800000,
                "y": 11,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532149200000,
                "y": 12,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532235600000,
                "y": 13,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532322000000,
                "y": 14,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532408400000,
                "y": 15,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532494800000,
                "y": 16,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532581200000,
                "y": 17,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532667600000,
                "y": 18,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532754000000,
                "y": 19,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532840400000,
                "y": 20,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1532926800000,
                "y": 21,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533013200000,
                "y": 22,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533099600000,
                "y": 23,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533186000000,
                "y": 24,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533272400000,
                "y": 25,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533358800000,
                "y": 26,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533445200000,
                "y": 27,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533531600000,
                "y": 28,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533618000000,
                "y": 29,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533704400000,
                "y": 30,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533790800000,
                "y": 31,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533877200000,
                "y": 32,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1533963600000,
                "y": 33,
              },
              Object {
                "extraMetrics": Array [],
                "series": "Count",
                "x": 1534050000000,
                "y": 34,
              },
            ],
          },
        ],
        "yAxisLabel": "Count",
      }
    }
  />
</svg>
`;
